src/RSet.php

```diff
@@ -223,6 +223,46 @@ public function addDate($date)
 		return $this;
 	}
 
+	/**
+	 * Remove an RDATE
+	 *
+	 * @param mixed $date a valid date representation or a \DateTime object
+	 * @return $this
+	 */
+	public function removeDate($date)
+	{
+		try {
+			$date_to_remove = RRule::parseDate($date);
+			$index = array_search($date_to_remove, $this->rdates);
+
+			if ( $index !== false ) {
+				unset($this->rdates[$index]);
+				$this->rdates = array_values($this->rdates);
```

Comment:
Why do you need this?

Reply:
This rekeys the array so that there are no gaps after an entry is removed. It may not be required but I wasn't sure about all the ways this array was used internally. I preferred that the internal array be left as if nothing had happened.

Reply:
Ok I don't think it's necessary. I tried removing it and the test passed all the same. But anyway it doesn't hurt so I'll merge this PR. Thank you.